src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,904 @@
+# coding=utf-8
+# Copyright 2023 Authors at City University of Hong Kong, Microsoft Cloud + AI,
+# The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch ICT model."""
+
+
+import math
+from typing import Optional, Tuple, Union
+
+import torch
+import torch.optim as optim
+import torch.utils.checkpoint
+from torch import nn
+
+from ...activations import ACT2FN
+from ...modeling_outputs import BaseModelOutput, ImageSuperResolutionOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from .configuration_ict import ICTConfig, ICTGuidedUpsamplerConfig, ICTTransformerConfig
+
+
+logger = logging.get_logger(__name__)
+
+# General docstring
+_CONFIG_FOR_DOC = "ICTConfig"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "sheonhan/ict-imagenet-32"
+_EXPECTED_OUTPUT_SHAPE = [1, 197, 768]  # TODO
```

Comment:
Not quite sure how to determine this shape

Reply:
If the authors have released more than one checkpoint for the model, we choose one (usually the base model) and use its preprocessing and configuration values to define the default parameters of `XXXImageProcessor` and `XXXConfig` classes. We also use this default model and corresponding repo on the hub ("huggingface/ict-imagenet-256") for the code snippets in the docs and for the model integration tests.

I think the authors have checkpoints for models trained on ImageNet, FFHQ and Places datasets and they say they keep the input/output size fixed for all models (256 x 256). ->` _EXPECTED_OUTPUT_SHAPE = [3, 256, 256]`